### DISCUSSION TITLE
Known Data Issues [Updated November 2025)

### DISCUSSION THREAD

**Discussion by rachelstephlee** [AllenNeuralDynamics/aind-dynamic-foraging-data-utils#58] [General]
# Historical Issues for animal behavior 

1. **Lick issue**

TLDR: 
Issue is present for any data collected prior to May 6, 2024. 
- Smart mice have figured this out and found a heuristic for learning the task
- This is found in all task types (coupled/uncoupled, baited/unbaited), at every stage. 

How to find issue: 
- Not every session has the issue, but please double check lick pattern
1. Check at ITI if there's an interlick interval peak around 1s or 2s
2. Check if there's a peak of lick rate around the start of no-lick delay.

Issue has been **fixed** for data collected after May 6, 2024. 
- After May, all mice that were trained in the previous set up had a major behavior performance drop because of longer delay time (an aversive thing for the mouse) for early licking. 
- Animals that have started training *after* May have better performance. 

Discussion here: 
[2024-04-30: AIND Behavior Meeting · AllenNeuralDynamics/aind-behavior-blog · Discussion #343](https://github.com/AllenNeuralDynamics/aind-behavior-blog/discussions/343)

Slides for problematic licking pattern:
[Presentation1.pptx](https://alleninstitute-my.sharepoint.com/:p:/g/personal/sue_su_alleninstitute_org/EVmC5RQfNYNMk8-uzm0Xz1wBNK5sNP1p1qM7NhyVghUdfQ?e=OaElLA)

2. **Motivation Issue**

TLDR: 
Issue is present in all data prior to July 4, 2024
- We have a shorter go cue, mouse were not able to hear the go cue because they were too old (and lose hearing for high frequency). 

How to find issue: 
- Data is potentially usable, as long as lick latency to go cue is normal (90% <500ms), and response rate is higher than 70% in the first 300 trials. 

Issue has been fixed after July 4th, 2024
- The fix was to increase the volume of go cue. 

Discussion here: 
https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/452

3. **significant delay between reward choice and reward delivery**

Description: 
Last friday (11/14) I noticed there was a significant delay between reward choice and reward delivery such that reward delivery was bleeding into the next trial. This resulted in a lot of delayed starts and early licking for my mouse (810889). I believe this is related to @bwtan-allen observation in the behavior performance channel. @ellahiltonvano and I confirmed this was also the case in 446 box 6A

This was attributed to the second stimuli PR . 

Discussion here: 
https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/1316

Fixed issue here: 
https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/1671

The PR should have been reverted by 11/18/2025. This should have impacted from 11/12/2025 (inclusive). 
- Per Bowen: I suspect some of the mice running on photometry rigs (6D, 7D, 8D, 1D, 2D, 3D) might have been affected before 11/11/25. It may be worth paying extra attention to the raw data.
- Rachel should check if, prior to wednesday 11/12, if any sessions on fiber rigs were run with the dev branch: 
```
"software": [
            {
               "name": "dynamic-foraging-task",
               "version": "behavior branch:main   commit ID:d01b82e86bc69691e3f8f4c92a50459b2a67e30d    version:1.6.54; metadata branch: main   commit ID:d01b82e86bc69691e3f8f4c92a50459b2a67e30d   version:1.6.54",
               "url": "https://github.com/AllenNeuralDynamics/dynamic-foraging-task.git",
               "parameters": {}
            }
```

# Pipeline Processing issues 

Any NWBs created prior to October 2024 are not to be trusted and have alignment issues. 

Check the pipeline and the merge history. Rachel's conservative estimate is: https://github.com/AllenNeuralDynamics/NWB-Packaging-FiberPhotometry-Base-Capsule/commit/ae0afd7776527a1781308696283fc1da2327f6e1
